moonstone/utils/df_split.py

```diff
@@ -0,0 +1,47 @@
+import pandas as pd
+from typing import Union
+
+from moonstone.core.module_base import BaseDF
+
+
+class DivideByGroup(BaseDF):
```

Comment:
I am not sure about what you want to do with this class? If the aim is to be able to split a dataframe based on another metadata dataframe, I think there must be a less complicated way.

Maybe the reason is different but when I need to loop through all data for all different group, here is what I usually do, considering a df, metadata_df and a group_col:

```python
# Concat metadata_df and df based on index
df = pd.concat([metadata_df[group_col], df], axis=1).dropna()

# Loop through different groups
groups = list(df[group_col].unique())
for group in groups:
    filtered_df = df[df[group_col] == group]
```

Reply:
Yes, I didn't use it in the end, it was really if you wanted to have more than one group in a dataframe. But of course you can just create another column and give it as the group_col instead

Reply:
Okay, then maybe we can get rid of it if we do not use it, we don't want to have part of the code which is not used